src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2854,31 +2854,68 @@ private uint getJitFlags(ref CORJIT_FLAGS flags, uint sizeInBytes)
             if (targetArchitecture == TargetArchitecture.ARM && !_compilation.TypeSystemContext.Target.IsWindows)
                 flags.Set(CorJitFlag.CORJIT_FLAG_RELATIVE_CODE_RELOCS);
 
-            if ((targetArchitecture == TargetArchitecture.X86
-                || targetArchitecture == TargetArchitecture.X64)
+            if (targetArchitecture == TargetArchitecture.X86)
+            {
+                flags.Set(InstructionSet.X86_SSE);
+                flags.Set(InstructionSet.X86_SSE2);
+#if !READYTORUN
+                // This list needs to match the list of intrinsics we can generate detection code for
+                // in HardwareIntrinsicHelpers.EmitIsSupportedIL.
+#else
+                // For ReadyToRun, this list needs to match up with the behavior of FilterNamedIntrinsicMethodAttribs
+                // In particular, that this list of supported hardware will not generate non-SSE2 safe instruction
+                // sequences when paired with the behavior in FilterNamedIntrinsicMethodAttribs
+                if (isMethodDefinedInCoreLib())
+#endif
+                {
+                    flags.Set(InstructionSet.X86_AES);
+                    flags.Set(InstructionSet.X86_PCLMULQDQ);
+                    flags.Set(InstructionSet.X86_SSE3);
+                    flags.Set(InstructionSet.X86_SSSE3);
+                    flags.Set(InstructionSet.X86_LZCNT);
 #if READYTORUN
-                && isMethodDefinedInCoreLib()
+                    flags.Set(InstructionSet.X86_SSE41);
+                    flags.Set(InstructionSet.X86_SSE42);
+                    flags.Set(InstructionSet.X86_POPCNT);
 #endif
-               )
+                }
+            }
+            else if (targetArchitecture == TargetArchitecture.X64)
```

Comment:
We could combine the x86 and x64 codepaths and just add the `*_X64` variants for x64.

Reply:
There is no guarantee that the X86 and X64 variants actually use the same constants. (It so happens they do at the moment, but that is one of the few actually negative consequences of this change, which is that the constants could become out of sync.)

Reply:
That is unfortunate, as it means we have a lot of duplicated code between the two of them (and that will likely grow over time).

Reply:
That does seem unfortunate. Does it really add a lot of complexity to share them?

Reply:
Well, it would be quite a bit of complexity, as the generator is pretty dumb. Mind you, the followon work is expected to delete this (the only logic which adds this duplication of effort) in favor of some string handling which will make all of this unified again. I hope to put together that PR by end of week, assuming I can get this in tonight or tomorrow morning, so I don't see much value in investing in a bit of code that should only live for about 48-96 hours before beginning active replacement.